collector/pkg/component/analyzer/pgftmetricanalyzer/pgft_analyzer.go

```diff
@@ -0,0 +1,144 @@
+	telemetry *component.TelemetryTools
+}
+
+func NewPgftMetricAnalyzer(cfg interface{}, telemetry *component.TelemetryTools, nextConsumers []consumer.Consumer) analyzer.Analyzer {
+	retAnalyzer := &PgftMetricAnalyzer{
+		consumers: nextConsumers,
+		telemetry: telemetry,
+	}
+	return retAnalyzer
+}
+
+func (a *PgftMetricAnalyzer) Start() error {
+	return nil
+}
+
+func (a *PgftMetricAnalyzer) ConsumableEvents() []string {
+	return []string{
+		constnames.PageFaultEvent,
+	}
+}
+
+// ConsumeEvent gets the event from the previous component
+func (a *PgftMetricAnalyzer) ConsumeEvent(event *model.KindlingEvent) error {
+	var dataGroup *model.DataGroup
+	var err error
+	if event.Name == constnames.PageFaultEvent {
+		dataGroup, err = a.generatePageFault(event)
+	}
+	if err != nil {
+		if ce := a.telemetry.Logger.Check(zapcore.DebugLevel, "Event Skip, "); ce != nil {
+			ce.Write(
+				zap.Error(err),
+			)
+		}
+		return nil
+	}
+	if dataGroup == nil {
+		return nil
+	}
+	var retError error
+	for _, nextConsumer := range a.consumers {
+		err := nextConsumer.Consume(dataGroup)
+		if err != nil {
+			retError = multierror.Append(retError, err)
+		}
+	}
+	return retError
+}
+
+func (a *PgftMetricAnalyzer) generatePageFault(event *model.KindlingEvent) (*model.DataGroup, error) {
+	labels, err := a.getPageFaultLabels(event)
+	if err != nil {
+		return nil, err
+	}
+
+	pgftMaj := event.GetUserAttribute("pgft_maj")
+	pgftMin := event.GetUserAttribute("pgft_min")
+	if pgftMaj == nil || pgftMin == nil {
+		return nil, fmt.Errorf("the page fault major or minor value is null %s", event.Name)
+	}
+	ptMaj := pgftMaj.GetUintValue()
+	ptMin := pgftMin.GetUintValue()
+
+	dataMaj := model.NewIntMetric(constnames.PgftMajorMetricName, int64(ptMaj))
+	dataMin := model.NewIntMetric(constnames.PgftMinorMetricName, int64(ptMin))
+	var dataSlice []*model.Metric
+	dataSlice = append(dataSlice, dataMaj)
+	dataSlice = append(dataSlice, dataMin)
+
+	return model.NewDataGroup(constnames.PgftMetricGroupName, labels, event.Timestamp, dataSlice...), nil
+}
+
+func getHostNameFromEnv() (string, error) {
+	value, ok := os.LookupEnv("MY_NODE_NAME")
+	if !ok {
+		return "unknown", fmt.Errorf("MY_NODE_NAME is not found in env variable which will be set unknown")
+	}
+	return value, nil
+}
+
+func (a *PgftMetricAnalyzer) getPageFaultLabels(event *model.KindlingEvent) (*model.AttributeMap, error) {
+
+	labels := model.NewAttributeMap()
+	ctx := event.GetCtx()
+	if ctx == nil {
+		return labels, fmt.Errorf("ctx is nil for event %s", event.Name)
+	}
+
+	threadinfo := ctx.GetThreadInfo()
+	if threadinfo == nil {
+		return labels, fmt.Errorf("threadinfo is nil for event %s", event.Name)
+	}
+
+	containerId := threadinfo.GetContainerId()
+
+	tid := (int64)(threadinfo.GetTid())
+	pid := (int64)(threadinfo.GetPid())
+
+	var localNodeName string
+	var err error
+	if localNodeName, err = getHostNameFromEnv(); err != nil {
+		return labels, fmt.Errorf("localName not found with %s", event.Name)
+	}
```

Comment:
Try to use a static string to save the HostName instead of calling os.GetEnv() every time

Reply:
OK, I put the method of getting localname into newPgftMetricAnanalyzer().